appcues/src/main/java/com/appcues/AppcuesViews.kt

```diff
@@ -0,0 +1,100 @@
+package com.appcues
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import android.widget.FrameLayout
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.ComposeView
+
+/**
+ * Base class for Appcues Compositions, Wraps an internal compose view and disable addView calls
+ * similarly to AbstractComposeView from compose.ui
+ */
+public abstract class AppcuesComposeView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : FrameLayout(context, attrs, defStyleAttr) {
+
+    private val composeView = ComposeView(context)
+
+    override fun getAccessibilityClassName(): CharSequence {
+        return javaClass.name
+    }
+
+    internal fun setComposition(composition: @Composable () -> Unit) {
+        composeView.setContent(composition)
+        super.addView(composeView)
+    }
+
+    internal fun clearComposition() {
+        super.removeAllViews()
+        composeView.setContent { }
+    }
+
+    // Below: enforce restrictions on adding child views to this ViewGroup
+    private fun checkAddView(child: View?) {
+        if (child != composeView) {
```

Comment:
the only child we want to add is the composeView

Reply:
Just an opinion, but I'd personally prefer the SDK not be opinionated here and simply provide the view with the understanding that if you mess with adding contents it is at your own peril (no exception throwing). Basically, not need any of these `checkAddView` calls and override safeguards. In practice, probably no one will do it and it won't matter either way - so it's harmless to include probably.